bridge/slack/handlers.go

```diff
@@ -90,248 +98,170 @@ func (b *Bslack) handleMatterHook(messages chan *config.Message) {
+					rmsg.Text = attach.Title + "\n"
+				}
+				rmsg.Text += attach.Text
+			} else {
+				rmsg.Text = attach.Fallback
+			}
 		}
-		return nil, fmt.Errorf("empty message and not a deleted message")
 	}
 
-	// save the attachments, so that we can send them to other slack (compatible) bridges
+	// Save the attachments, so that we can send them to other slack (compatible) bridges.
 	if len(ev.Attachments) > 0 {
 		rmsg.Extra[sSlackAttachment] = append(rmsg.Extra[sSlackAttachment], ev.Attachments)
 	}
 
-	// if we have a file attached, download it (in memory) and put a pointer to it in msg.Extra
+	// If we have files attached, download them (in memory) and put a pointer to it in msg.Extra.
 	for _, f := range ev.Files {
-		err := b.handleDownloadFile(&rmsg, &f)
+		err := b.handleDownloadFile(rmsg, &f)
 		if err != nil {
-			b.Log.Errorf("download failed: %s", err)
+			b.Log.Errorf("Could not download incoming file: %#v", err)
 		}
 	}
-
-	return &rmsg, nil
 }
 
-// skipMessageEvent skips event that need to be skipped :-)
-func (b *Bslack) skipMessageEvent(ev *slack.MessageEvent) bool {
-	if ev.SubType == sChannelLeave || ev.SubType == sChannelJoin {
-		return b.GetBool(noSendJoinConfig)
-	}
-
-	// ignore pinned items
-	if ev.SubType == sPinnedItem || ev.SubType == sUnpinnedItem {
-		return true
-	}
+var commentRE = regexp.MustCompile(`.*?commented: (.*)`)
 
-	// do not send messages from ourself
-	if b.GetString(outgoingWebhookConfig) == "" && b.GetString(incomingWebhookConfig) == "" && ev.Username == b.si.User.Name {
-		return true
+// handleDownloadFile handles file download
+func (b *Bslack) handleDownloadFile(rmsg *config.Message, file *slack.File) error {
+	// Only download a file if it is not in the cache or if it has been entered more than a minute ago.
+	if ts, ok := b.cache.Get("file" + file.ID); ok && time.Since(ts.(time.Time)) > time.Minute {
+		return nil
+	} else if ts, ok = b.cache.Get("filename" + file.Name); ok && time.Since(ts.(time.Time)) > 10*time.Second {
+		return nil
 	}
 
-	// skip messages we made ourselves
-	if len(ev.Attachments) > 0 {
-		if ev.Attachments[0].CallbackID == "matterbridge_"+b.uuid {
-			return true
-		}
+	// Check that the file is neither too large nor blacklisted.
+	if err := helper.HandleDownloadSize(b.Log, rmsg, file.Name, int64(file.Size), b.General); err != nil {
+		b.Log.WithError(err).Infof("Skipping download of incoming file.")
+		return nil
 	}
 
-	if !b.GetBool(editDisableConfig) && ev.SubMessage != nil && ev.SubMessage.ThreadTimestamp != ev.SubMessage.Timestamp {
-		// it seems ev.SubMessage.Edited == nil when slack unfurls
-		// do not forward these messages #266
-		if ev.SubMessage.Edited == nil {
-			return true
-		}
+	// Actually download the file.
+	data, err := helper.DownloadFileAuth(file.URLPrivateDownload, "Bearer "+b.GetString(tokenConfig))
+	if err != nil {
+		return fmt.Errorf("download %s failed %#v", file.URLPrivateDownload, err)
 	}
 
-	for _, f := range ev.Files {
```

Comment:
This piece of logic has been removed from the `skipMessageEvent()` method (which as moved higher-up in the file).

As far as my understanding goes the old logic below could have resulted in a message with multiple attachments being skipped if only one of the files was present and fresh in the cache although others were not.

Hence I've moved the verification to be done within `handleDownloadFile()` which is called separately for each file.